### PR TITLE
Move adding background media to the toolbar

### PR DESCRIPTION
**Pull request by spacedmonkey**
Fixes: #2699

Steps to test: 
1. Create a new story. 
1. Click on media icon in top toolbar. 
1. Dropdown should appear with following options.
   - Insert image (inserts block)
   - Insert video (inserts block)
   - Insert background image (opens media modal)
   - Insert background video (opens media modal)
1. Insert image / video should these blocks into the current page. 
1. Insert background image / video, should open a media picker dialog and should allow for upload / setting of background.

### CODE REVIEW

File: assets/src/stories-editor/components/index.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ export { default as FontFamilyPicker } from './font-family-picker';
 export { default as RotatableBox } from './rotatable-box';
 export { default as PreviewPicker } from './preview-picker';
 export { default as Inserter } from './inserter';
+export { default as MediaInsider } from './media-inserter';
```
Comment (swissspidy) [inline assets/src/stories-editor/components/index.js]: Typo: should be `MediaInserter`

Comment (spacedmonkey) [reply assets/src/stories-editor/components/index.js]: Good spot. Resolved in https://github.com/ampproject/amp-wp/pull/3093/commits/7f1f2b9155f9869acb0ae027c8a31512ca8d589e

File: assets/src/stories-editor/blocks/amp-story-page/edit.js [hunk 2]
```diff
@@ -314,7 +314,6 @@ class PageEdit extends Component {
 				<InspectorControls>
 					<PanelColorSettings
 						title={ __( 'Background Color', 'amp' ) }
-						initialOpen={ false }
```
Comment (swissspidy) [inline assets/src/stories-editor/blocks/amp-story-page/edit.js]: Can we do the same for regular blocks as well?

This is bugging me a bit:

https://github.com/ampproject/amp-wp/blob/c894d9a6bdb4d8d2ccf73d1ef58f8a73a6b76e1a/assets/src/stories-editor/components/higher-order/with-amp-story-settings.js#L471-L486

**Comment by swissspidy** [comment]
Would be nice if we could align the icons here:

Screenshot 2019-08-27 at 12 20 17

Compare it with the block toolbar menu:

Screenshot 2019-08-27 at 12 22 15

The icon color doesn't seem to match either

File: assets/src/stories-editor/blocks/amp-story-page/edit.js (hunk 2 above)
Comment (spacedmonkey) [reply assets/src/stories-editor/blocks/amp-story-page/edit.js]: This should be another issue / PR

File: assets/src/stories-editor/components/media-inserter/index.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+		{
+			title: __( 'Insert Background Image', 'amp' ),
+			icon: 'format-image',
+			onClick: () => mediaPicker( __( 'Select or Upload Media', 'amp' ), 'image', updateBlock ),
+			disabled: ! showInserter,
+		},
+		{
+			title: __( 'Insert Background Video', 'amp' ),
+			icon: 'media-video',
+			onClick: () => mediaPicker( __( 'Select or Upload Media', 'amp' ), 'video', updateBlock ),
```
Comment (spacedmonkey) [inline assets/src/stories-editor/components/media-inserter/index.js]: Why were these changed? The wording before was more clear.

File: assets/src/stories-editor/components/media-inserter/index.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+	canInsertBlockType: PropTypes.func.isRequired,
+	showInserter: PropTypes.bool.isRequired,
+};
+
+const mediaPicker = ( dialogTitle, mediaType, updateBlock ) => {
+	// Create the media frame.
+	const fileFrame = wp.media( {
+		title: dialogTitle,
+		button: {
+			text: __( 'Select', 'amp' ),
```
Comment (spacedmonkey) [inline assets/src/stories-editor/components/media-inserter/index.js]: Use a background is much clearer here.

**Comment by spacedmonkey** [comment]
> Would be nice if we could align the icons here:
> 
> Screenshot 2019-08-27 at 12 20 17
> 
> Compare it with the block toolbar menu:
> 
> Screenshot 2019-08-27 at 12 22 15
> 
> The icon color doesn't seem to match either

@swissspidy Fixed in https://github.com/ampproject/amp-wp/pull/3093/commits/10da2ae551ff6eebfba6f5697ba2afa8939dd956 

Screenshot from 2019-08-27 15-17-12

**Comment by spacedmonkey** [comment]
I believe this PR is good to go. It has e2e test and lints are passing.

**Comment by miina** [comment]
Thanks for the PR, works well, much easier to handle the Media now!

One small styling comment -- all the other dropdowns in the Toolbar seem to open from either right or left side, could we make this Dropdown show like that as well for consistency? E.g. the "More options" and "More tools & options" etc. 
For example, the "More options" Dropdown for `RichText` opening from the right corner:
Screenshot 2019-08-27 at 17 11 54

The custom Media Inserter opening from the middle:
Screenshot 2019-08-27 at 17 12 11

File: assets/src/stories-editor/components/media-inserter/index.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+			const mediaUrl = media.url;
+			const poster = VIDEO_BACKGROUND_TYPE === mediaType && media.image && media.image.src !== media.icon ? media.image.src : undefined;
+			updateBlockAttributes( clientId, {
+				mediaUrl,
+				mediaId: media.id,
+				mediaType,
+				mediaAlt,
+				poster,
+			} );
+			selectBlock( clientId );
```
Comment (miina) [inline assets/src/stories-editor/components/media-inserter/index.js]: This seems to be largely matching the logic from `onSelectMedia` of the Page block -- could we create a helper function instead for the matching part and move it to `helpers` to reduce duplication?

Comment (spacedmonkey) [reply assets/src/stories-editor/components/media-inserter/index.js]: Created a new helper function called `processMedia`.

File: assets/src/stories-editor/components/media-inserter/index.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+import { DropdownMenu } from '@wordpress/components';
+import { compose, ifCondition } from '@wordpress/compose';
+import { __, sprintf } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import { processMedia } from '../../helpers';
+
+const POPOVER_PROPS = {
+	position: 'bottom right',
```
Comment (spacedmonkey) [inline assets/src/stories-editor/components/media-inserter/index.js]: Based on @miina this now is styled bottle right.

File: assets/src/stories-editor/blocks/amp-story-page/edit.js [hunk 7]
```diff
@@ -43,7 +43,7 @@ import {
 	getCallToActionBlock,
 	getUniqueId,
 	uploadVideoFrame,
-	getPosterImageFromFileObj,
+	getPosterImageFromFileObj, processMedia,
```
Comment (miina) [inline assets/src/stories-editor/blocks/amp-story-page/edit.js]: ^ This could be in a new line for consistency.

Comment (swissspidy) [reply assets/src/stories-editor/blocks/amp-story-page/edit.js]: I really need to find an ESLint rule at some point to catch these kind of things 🤔

Comment (spacedmonkey) [reply assets/src/stories-editor/blocks/amp-story-page/edit.js]: There was nothing in the linter for this. 
FIxed in https://github.com/ampproject/amp-wp/pull/3093/commits/62aa7fbdd2203f22a04ef77491dedafac3bfeb13
